src/main/java/com/tenx/ms/retail/product/rest/ProductController.java

```diff
@@ -0,0 +1,142 @@
+package com.tenx.ms.retail.product.rest;
+
+import com.tenx.ms.commons.rest.RestConstants;
+import com.tenx.ms.commons.rest.dto.ResourceCreated;
+import com.tenx.ms.retail.product.rest.dto.Product;
+import com.tenx.ms.retail.product.service.ProductService;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiResponses;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiParam;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.http.HttpStatus;
+import org.springframework.transaction.TransactionSystemException;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.validation.ConstraintViolationException;
+import java.io.IOException;
+import java.util.List;
+
+/**
+ * Created by goropeza on 26/08/16.
+ */
+@Api(value = "product", description = "Product API")
+@RestController("productControllerV1")
+@RequestMapping(RestConstants.VERSION_ONE + "/products")
+public class ProductController {
+
+    @Autowired
+    private ProductService productService;
+
+    @ApiOperation(value = "Return all products in a store")
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = "Product list returned successfully"),
+        @ApiResponse(code = 404, message = "Store couldn't be found"),
+        @ApiResponse(code = 500, message = "Internal server error")}
+    )
+    @RequestMapping(value = {"/{storeId:\\d+}"}, method = RequestMethod.GET)
+    public List<Product> findProductsInStore(@ApiParam(name = "storeId", value = "Store id")
+                              @PathVariable() Long storeId) {
+        return productService.findAllProductsByStore(storeId);
+    }
+
+    @ApiOperation(value = "Find a product in a store by product id")
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = "Product list returned successfully"),
+        @ApiResponse(code = 404, message = "Store or product couldn't be found"),
+        @ApiResponse(code = 500, message = "Internal server error")}
+    )
+    @RequestMapping(value = {"/{storeId:\\d+}"}, params={"id"}, method = RequestMethod.GET)
```

Comment:
RequestParam named id seems too generic. By looking at the url, the client is not sure whether that id is store id or product id. Recommend to use more descriptive param name